src/services/ExperianService.ts

```diff
@@ -0,0 +1,124 @@
+import { DynamoDBDocument, GetCommand, PutCommand } from "@aws-sdk/lib-dynamodb";
+import { randomUUID } from "crypto";
+
+export class ExperianService {
+	readonly logger: Logger;
+
+	private static instance: ExperianService;
+
+	private readonly dynamo: DynamoDBDocument;
+
+	readonly experianTokenTableName: string;
+
+	readonly experianBaseUrl: string;
+
+	constructor(logger: Logger, experianBaseUrl: string, dynamoDbClient: DynamoDBDocument, experianTokenTableName: string) {
+    	this.logger = logger;
+		this.dynamo = dynamoDbClient;
+		this.experianBaseUrl = experianBaseUrl;
+		this.experianTokenTableName = experianTokenTableName;
+	}
+
+	static getInstance(logger: Logger, experianBaseUrl: string, dynamoDbClient: DynamoDBDocument, experianTokenTableName: string): ExperianService {
+    	if (!ExperianService.instance) {
+    		ExperianService.instance = new ExperianService(logger, experianBaseUrl, dynamoDbClient, experianTokenTableName);
+    	}
+    	return ExperianService.instance;
+	}
+
+	async generateExperianToken(clientUsername: string, clientPassword: string, clientId: string, clientSecret: string): Promise<ExperianTokenResponse> {
+    	this.logger.info({ message: `Checking ${this.experianTokenTableName} for valid token` });
+		const token = await this.checkExperianToken();
+		if (token) {
+			return token;
+		} else {		
+			this.logger.info("requestExperianToken - no valid token - trying to generate new Experian token");
+			try {
+				const params = {
+					username: clientUsername,
+					password: clientPassword,
+					client_id: clientId,
+					client_secret: clientSecret,
+				};
+
+				const config: AxiosRequestConfig<any> = {
+    				headers: {
+    					"Content-Type": "application/json",
+						"X-Correlation-Id": randomUUID(),
+						"X-User-Domain": "cabinetofficegds.com",
+    				},
+				};
+
+				const { data }: { data: ExperianTokenResponse } = await axios.post(
+					`${this.experianBaseUrl}${Constants.EXPERIAN_TOKEN_ENDPOINT_PATH}`,
+					params,
+					config,
+				);
+				this.logger.info("Received response from experian token endpoint");
+				await this.saveExperianToken(data);
+				return data;
+			} catch (error: any) {
+				this.logger.error({ message: "Error generating experian token", statusCode: error?.response?.status, messageCode: MessageCodes.FAILED_GENERATING_EXPERIAN_TOKEN });
+				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error generating experian token");
+			}
+		}
+	}
+
+	async saveExperianToken(token : ExperianTokenResponse): Promise<void> {
+		this.logger.info({ message: `Updating ${this.experianTokenTableName} with new Experian token` });
+
+		const putExperianTokenCommand = new PutCommand({
+			TableName: this.experianTokenTableName,
+			Item: { 
+				id: "1",
+				issued_at: token.issued_at,
+				expires_in: token.expires_in,
+				token_type: token.token_type,
+				access_token: token.access_token,
+				refresh_token: token.refresh_token,
```

Comment:
We can remove refresh_token: token.refresh_token as it will never be used

Reply:
Added new model for stored token, removed refresh_token field from DB storage